### ISSUE TITLE
Validate schema extractor syntax

### ISSUE THREAD

**Issue by rvansa** [Hyperfoil/Horreum#40]
Just check `SELECT jsonb_path_query_first('{}', path)` to see if the syntax is correct when saving the extractor.

**Comment by JacksonIsaac** [comment]
Working on this :)

**Comment by JacksonIsaac** [comment]
I am currently looking at SchemaService.java#L274. Is this the correct place to add this check ? I am still trying to figure out how this syntax validation will be done.

I also assume that the check has to be done on extractor variable (line 316) ?

**Comment by JacksonIsaac** [comment]
I haven't used Horreum, but just tried to read the code to understand what's going on. So let me know what you had in mind, and I'll follow up from there.

**Comment by rvansa** [comment]
@JacksonIsaac As I've said above, a separate REST endpoint in [SchemaService](https://github.com/Hyperfoil/Horreum/blob/master/src/main/java/io/hyperfoil/tools/horreum/api/SchemaService.java) that would run `SELECT jsonb_path_query_first('{}', path)` and see if that throws any error, and report that back to Horreum.

On the UI side that would be [Schema.tsx](https://github.com/Hyperfoil/Horreum/blob/master/webapp/src/domain/schemas/Schema.tsx) where you'd fire the request 1 second after jsonpath change.

I think that the best first step would definitely be getting Horreum running locally to see the UI for yourselves. If you're using `podman` instead of `docker`, I've just created a compatible definition for that as well - please see README.

**Comment by rvansa** [comment]
@JacksonIsaac Hi, any progress? Have you managed to set up Horreum or do you have any issues?

**Comment by JacksonIsaac** [comment]
Hi @rvansa, I was able to setup Horreum and access the UI. I was looking at adding a @POST api in SchemaService (called maybe validate_schema), but I wasn't sure how the sql query would execute from there

**Comment by rvansa** [comment]
@JacksonIsaac SchemaService has injected `EntityManager em` and `AgroalDataSource dataSource` fields; You can either run a native query through the former one or use JDBC directly on the latter. Since the query does not access any data nor modifies anything you **do not** need to wrap it with
```java
try (CloseMe h = sqlService.withRoles(em, identity)) { ... }
```
as other DB calls in the class do.

**Comment by rvansa** [comment]
@JacksonIsaac Was ^ helpful? If you're still unsure, you can reach me online on hyperfoil.zulipchat.com

**Comment by rvansa** [comment]
Implemented in 54eaa76a49f6de36ac040540c08a7d84c1980858
